### PR TITLE
Create Validate interface for inband code update

### PR DESCRIPTION
**Pull request by sagisin**
This method will allow for validation to occur during inband code updates. It will ensure that the User Access Key, Signature, and Minimum Ship Level requirements are met before the update proceeds.

### CODE REVIEW

**Comment by geissonator** [comment]
add to approvelist

**Comment by anoo1** [comment]
CI error seems like a network issue, retriggering.

```
W: Failed to fetch http://ports.ubuntu.com/ubuntu-ports/dists/kinetic/InRelease  Cannot initiate the connection to ports.ubuntu.com:80 (2620:2d:4000:1::19). - connect (101: Network is unreachable
```

**Comment by anoo1** [comment]
jenkins run tests please

**Review by anoo1 — CHANGES_REQUESTED** [review]
Also add your name to the commit msg, and you can remove Zami as author since this has changed so much since her initial attempt for these changes.

File: yaml/xyz/openbmc_project/Software/LID.errors.yaml
[the remark this answers is not in the document]
Comment (sagisin) [reply yaml/xyz/openbmc_project/Software/LID.errors.yaml:1]: done

File: gen/xyz/openbmc_project/Software/LID/meson.build [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Software/LID__cpp'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Software/LID.interface.yaml',  ],
+    output: [ 'error.cpp', 'error.hpp', 'server.cpp', 'server.hpp', 'client.hpp',  ],
+    command: [
```
[the remark this answers is not in the document]
Comment (sagisin) [reply gen/xyz/openbmc_project/Software/LID/meson.build:7]: done

File: gen/xyz/openbmc_project/Software/meson.build [hunk 2]
```diff
@@ -89,6 +89,20 @@ generated_others += custom_target(
     ],
 )
 
+subdir('LID')
+generated_others += custom_target(
+    'xyz/openbmc_project/Software/LID__markdown'.underscorify(),
+    input: [ '../../../../yaml/xyz/openbmc_project/Software/LID.interface.yaml',  ],
+    output: [ 'LID.md' ],
+    command: [
```
[the remark this answers is not in the document]
Comment (sagisin) [reply gen/xyz/openbmc_project/Software/meson.build:98]: done

File: yaml/xyz/openbmc_project/Software/LID.interface.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+description: >
+    An interface to validate marker LID before starting inband code update.
```
[the remark this answers is not in the document]
Comment (sagisin) [reply yaml/xyz/openbmc_project/Software/LID.interface.yaml]: done

File: yaml/xyz/openbmc_project/Software/LID.interface.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+methods:
+    - name: Validate
+      descripton: >
+          Validate file in staging directory given the lid value 7.
```
[the remark this answers is not in the document]
Comment (sagisin) [reply yaml/xyz/openbmc_project/Software/LID.interface.yaml]: done

**Review by anoo1 — APPROVED** [review]

**Pull request merged**